arangod/GeneralServer/HttpCommTask.cpp

```diff
@@ -334,17 +334,24 @@ void HttpCommTask<T>::checkVSTPrefix() {
                       asio_ns::transfer_at_least(11), std::move(cb));
 }
 
-template <SocketType T>
-bool HttpCommTask<T>::checkHttpUpgrade() {
-  // TODO
-  return true;
-}
-
 template <SocketType T>
 void HttpCommTask<T>::processRequest() {
   TRI_ASSERT(_request);
   this->_protocol->timer.cancel();
 
+  // we may have gotten an H2 Upgrade request
+  if (ADB_UNLIKELY(_parser.upgrade)) {
+    bool found;
+    std::string const& h2 = _request->header("upgrade");
+    std::string const& settings = _request->header("http2-settings", found);
+    if (h2 == "h2c" && found && !settings.empty()) {
```

Comment:
we just check if `settings` is not empty here, but the actual content of `settings` is not looked into. I guess this is intentional for now, but would like you to confirm this.

Reply:
this is according to spec, everything else is left to the HTTP2 library